src/lib/gadgets/bitwise.ts

```diff
@@ -3,7 +3,44 @@ import { Field as Fp } from '../../provable/field-bigint.js';
 import { Field } from '../field.js';
 import * as Gates from '../gates.js';
 
-export { xor };
+export { xor, not };
+
+function not(a: Field, length: number) {
+  // check that input length is positive
+  assert(length > 0, `Input length needs to be positive values.`);
+
+  // Check that length does not exceed maximum field size in bits
+  assert(
+    length <= Field.sizeInBits(),
+    `Length ${length} exceeds maximum of ${Field.sizeInBits()} bits.`
+  );
+
+  // obtain pad length until the length is a multiple of 16 for n-bit length lookup table
+  let padLength = Math.ceil(length / 16) * 16;
+
+  // handle constant case
+  if (a.isConstant()) {
+    let max = 1n << BigInt(padLength);
+    assert(
+      a.toBigInt() < max,
+      `${a.toBigInt()} does not fit into ${padLength} bits`
+    );
+    return new Field(Fp.not(a.toBigInt()));
+  }
+
+  // create a bitmask with all ones
+  let allOnesF = new Field(BigInt(2 ** length - 1));
```

Comment:
(not so) fun fact: these all return different values

```ts
length = 64
let a = 2 ** length - 1;
console.log(a)
console.log(a.toString(2))

let b = 2n ** BigInt(length) - 1n;
console.log(b)
console.log(b.toString(2))

let c = BigInt(2 ** 64 - 1);
console.log(c)
console.log(c.toString(2))
```

```sh
18446744073709552000 
"10000000000000000000000000000000000000000000000000000000000000000" 

18446744073709551615 
"1111111111111111111111111111111111111111111111111111111111111111" 

8446744073709551616 
"10000000000000000000000000000000000000000000000000000000000000000" 

```

so I think we should go with `2n ** BigInt(length) - 1n`